common/model-views.cpp

```diff
@@ -4183,6 +4234,133 @@ namespace rs2
         return keep_showing;
     }
 
+    bool device_model::prompt_trigger_camera_accuracy_health(ux_window& window, viewer_model& viewer, std::string& error_message)
+    {
+        // This process is built from a 2 stages windows, first a yes/no window and then a process window
+        bool keep_showing = true;
+        bool yes_was_chosen = false;
+
+        switch (cah_model.cah_state)
+        {
+        case camera_accuracy_health_model::model_state_type::TRIGGER_MODAL:
+        {
+            const std::string message_text("Camera Accuracy Health will ensure you get the highest accuracy from your camera.\n\n"
+                                            "This process may take several minutes and requires special setup to get good results.\n"
+                                            "While it is working, the viewer will not be usable.\n\n"
+                                            "Are you sure you want to continue?");
+            if (yes_no_dialog("Camera Accuracy Health", message_text, yes_was_chosen, window, error_message))
+            {
+                if (yes_was_chosen)
+                {
+                    bool is_depth_streaming = std::any_of(subdevices.begin(), subdevices.end(), [](const std::shared_ptr<subdevice_model>& sm) { return sm->streaming && sm->s->as<depth_sensor>(); });
+                    bool is_color_streaming = std::any_of(subdevices.begin(), subdevices.end(), [](const std::shared_ptr<subdevice_model>& sm) { return sm->streaming && sm->s->as<color_sensor>(); });
+
+                    auto itr = std::find_if(subdevices.begin(), subdevices.end(), [](std::shared_ptr<subdevice_model> sub)
+                    {
+                        if (sub->s->as<depth_sensor>())
+                            return true;
+                        return false;
+                    });
+
+
+                    if (is_depth_streaming && is_color_streaming && itr != subdevices.end())
+                    {
+                        auto sd = *itr;
+                        sd->s->set_option(RS2_OPTION_TRIGGER_CAMERA_ACCURACY_HEALTH, 1.0f);
+                        device_calibration dev_cal(dev);
+                        // Register AC status change callback
+                        dev_cal.register_calibration_change_callback([&](rs2_calibration_status cal_stts)
+                        {
+                            cah_model.calib_stts = cal_stts;
+                            if (cah_model.calib_stts == RS2_CALIBRATION_STARTED || cah_model.calib_stts == RS2_CALIBRATION_SPECIAL_FRAME)
+                            {
+                                cah_model.cah_state = camera_accuracy_health_model::model_state_type::PROCESS_MODAL;
+                            }
+                        });
+
+                    }
+                    else
+                    {
+                        std::stringstream ss;
+                        ss << "Can not trigger Camera Accuracy Health, required streams not active";
```

Comment:
Instead of saying 'required streams not active":

Camera Accuracy Health cannot be triggered: both depth & RGB streams must be active

Reply:
Will update